schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -1,549 +1,835 @@
-                "createOption": {
-                    "type": "string"
-                }
+        "type": "object"
+      },
+      "ipConfiguration": {
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "properties": {
+            "subnet": {
+              "$ref": "#/definitions/id"
             },
-            "type": "object",
-            "required": [
-                "name",
-                "vhd",
-                "createOption"
-            ]
+            "loadBalancerBackendAddressPools": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/id"
+              }
+            }
+          },
+          "type": "object",
+          "required": [
+            "subnet"
+          ]
         },
-        "vhdUri": {
-            "properties": {
-                "uri": {
-                    "type": "string"
-                }
+        "type": "object",
+        "required": [
+          "name",
+          "properties"
+        ]
+      },
+      "networkInterfaceConfiguration": {
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "properties": {
+            "primary": {
+              "type": "boolean"
             },
-            "type": "object",
-            "required": [
-                "uri"
-            ]
+            "ipConfigurations": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ipConfiguration"
+              }
+            }
+          },
+          "type": "object",
+          "required": [
+            "ipConfigurations"
+          ]
         },
-        "dataDisk": {
+        "type": "object",
+        "required": [
+          "name",
+          "properties"
+        ]
+      },
+      "virtualMachineScaleSetNetworkProfile": {
+        "properties": {
+          "networkInterfaceConfigurations": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/networkInterfaceConfiguration"
+            }
+          }
+        },
+        "type": "object",
+        "required": [
+          "networkInterfaceConfigurations"
+        ]
+      },
+      "virtualMachineProfile": {
+        "properties": {
+          "osProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetOsProfile"
+          },
+          "storageProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetStorageProfile"
+          },
+          "extensionProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetExtensionProfile"
+          },
+          "networkProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetNetworkProfile"
+          }
+        },
+        "type": "object",
+        "required": [
+          "osProfile",
+          "storageProfile",
+          "extensionProfile",
+          "networkProfile"
+        ]
+      },
+      "virtualMachineScaleSets": {
```

Comment:
If this is a new resource type can be added in a template, please add reference into "2015-01-01\deploymentTemplate.json" Ln 40 and "2014-04-01-preview\deploymentTemplate.json" Ln 43.